api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -3311,4 +3314,10 @@ public void mergePatients_shouldFailIfNotPreferredPatientHasUnvoidedOrders() thr
 		Patient notPreferredPatient = patientService.getPatient(7);
 		patientService.mergePatients(preferredPatient, notPreferredPatient);
 	}
+
+	
+	@Test(expected = APIException.class)
+	public void processDeath_shouldThrowAPIExceptionIfPatientIsNull() throws Exception{
+		patientService.processDeath(null, new Date(), new Concept(), "unknown");
```

Comment:
Future efforts will be spent in the business logic direction, however the purpose here is to test robustness as software should fail safely.